grails-app/controllers/org/pih/warehouse/api/PartialReceivingApiController.groovy

```diff
@@ -194,7 +194,7 @@ class PartialReceivingApiController {
                 if (!partialReceiptItem || newLine) {
                     partialReceiptItem = new PartialReceiptItem()
                     partialReceiptItem.shipmentItem = ShipmentItem.get(shipmentItemId)
-                    partialReceiptItem.product = shipmentItemMap.get("product.id") ? Product.load(shipmentItemMap.get("product.id")) : null
+                    partialReceiptItem.product = Product.get(shipmentItemMap?.product?.id)
```

Comment:
is there a reason you changed it from `load()` to `get()`? I assume maybe `load()` was used there on purpose, because of performance - it doesn't hit the DB immedietaly as `get()` and you are inside `.each` loop.

Reply:
I was not aware of `load()` not making aN SQL query, I always thought that the difference between **get** and **load** was that load throws and exception when a record is not found in db while get just returns a null.

I have found such a table on [stack](https://stackoverflow.com/a/32964515) explaining when to use which
![image](https://github.com/openboxes/openboxes/assets/35233347/f8c78bf9-b6eb-4512-a9dc-6de937d757f4)

thanks, @kchelstowski for pointing this out, maybe it will be a good idea to change it load to avoid future performance complications, however small they could be.

Reply:
I'm fine we either approach, but load seems to be more tempting. I think we should use it more.

Reply:
I use load when I need to set a foreign key. 

The fact that there's a ShipmentItem.get() invocation right before this means that the Product.load() was either arbitrary. 

Please note that the following line 
```
partialReceiptItem.product = shipmentItemMap.get("product.id") ? Product.load(shipmentItemMap.get("product.id")) : null
```
is different from this line
```
partialReceiptItem.product = Product.get(shipmentItemMap?.product?.id)
```
in more ways than just the get vs load. You may have inadvertently caused a regression with this change in the case where a product ID is not specified i.e. do we know for sure that product.id will always be not null? Otherwise, we're going to get a loud exception here instead of a null product reference.

I would recommend going back to the logic "if key exists, look up Product by key, otherwise return null"

cc @drodzewicz @awalkowiak @kchelstowski

Reply:
@jmiranda are you sure it works the way you described? I don't remind myself `get()` throwing an error if I provide `null` as an argument.
I checked it via console:
![Screenshot from 2023-07-07 08-36-57](https://github.com/openboxes/openboxes/assets/93163821/dece5b51-176f-4b2c-ac7c-d4bfe9f64778)

For those, the only that will throw the exception will be `load()` called with an id that do not exist (but with `null` it surprisingly works fine)

Reply:
@kchelstowski I was just pointing out that this code 
```
def variable = checkSomething ? doSomething : null
```
is not equivalent to 
```
def variable = doSomething
```
and that there might be side effects to using the latter.